examples/index.ts

```diff
@@ -92,6 +92,7 @@ main()
   .then(() => {
     console.log('success!!');
   })
-  .catch(e => {
-    console.error('failed to get from cache', e);
+  .catch((e: Error) => {
+    console.error(`failed to get from cache ${e.message}`);
+    throw e;
```

Comment:
thank you for fixing this!  I have noticed that we are missing the throws for these in some places and it makes it really confusing to debug :)

Reply:
I should make them consistent and all throw?

Reply:
I was talking about in other repos where we use TS for infra, I'm not aware of whether there are any more missing in this repo.  If you audited and they all look good in this repo, great!